jgrapht-core/src/main/java/org/jgrapht/alg/matching/MaximumWeightBipartiteMatching.java

```diff
@@ -75,210 +102,286 @@ public MaximumWeightBipartiteMatching(Graph<V, E> graph, Set<V> partition1, Set<
-        for (E edge : bipartiteMatching) {
-            weight += graph.getEdgeWeight(edge);
+        // empty graph
+        if (graph.edgeSet().isEmpty()) {
+            return new MatchingImpl<>(graph, matching, matchingWeight.doubleValue());
         }
-        return new MatchingImpl<>(graph, bipartiteMatching, weight);
-    }
 
-    private void initializeVerticesAndEdges()
-    {
-        for (V vertex : graph.vertexSet()) {
-            if (isTargetVertex(vertex)) {
-                hasVertexBeenProcessed.put(vertex, true);
-                setVertexWeight(vertex, (long) 0);
-            } else {
-                hasVertexBeenProcessed.put(vertex, false);
-                setVertexWeight(vertex, maximumWeightOfEdgeIncidentToVertex(vertex));
+        // initialize
+        pot = new HashMap<>();
+        dist = new HashMap<>();
+        matchedEdge = new HashMap<>();
+        heap = new GenericFibonacciHeap<>(comparator);
+        isInHeap = new HashSet<>();
+        nodeInHeap = new HashMap<>();
+        pred = new HashMap<>();
+        graph.vertexSet().stream().forEach(v -> {
+            pot.put(v, BigDecimal.ZERO);
+            nodeInHeap.put(v, null);
+            pred.put(v, null);
+            dist.put(v, BigDecimal.ZERO);
+        });
+
+        // run simple heuristic
+        simpleHeuristic();
+
+        // augment to optimality
+        for (V v : partition1) {
+            if (!matchedEdge.containsKey(v)) {
+                augment(v);
             }
         }
 
-        for (E edge : graph.edgeSet()) {
-            isEdgeMatched.put(edge, false);
-        }
+        return new MatchingImpl<>(graph, matching, matchingWeight.doubleValue());
     }
 
-    private long maximumWeightOfEdgeIncidentToVertex(V vertex)
+    /**
+     * Get the vertex potentials.
+     * 
+     * <p>
+     * This is a tight non-negative potential function which proves the optimality of the maximum
+     * weight matching. See any standard textbook about linear programming duality.
+     * 
+     * @return the vertex potentials
+     */
+    public Map<V, BigDecimal> getPotentials()
     {
-        long maxWeight = 0;
-        for (E edge : graph.edgesOf(vertex)) {
-            if (graph.getEdgeWeight(edge) > maxWeight) {
-                maxWeight = (long) graph.getEdgeWeight(edge);
-            }
+        if (pot == null) {
+            return Collections.emptyMap();
+        } else {
+            return Collections.unmodifiableMap(pot);
         }
-        return maxWeight;
     }
 
-    private boolean isSourceVertex(V vertex)
+    /**
+     * Get the weight of the matching.
+     * 
+     * @return the weight of the matching
+     */
+    public BigDecimal getMatchingWeight()
     {
-        return partition1.contains(vertex);
+        return matchingWeight;
     }
 
-    private boolean isTargetVertex(V vertex)
+    /**
+     * Get the matching edges.
+     * 
+     * @return the matching edges
+     */
+    public Set<E> getMatchingEdges()
```

Comment:
I understand why you're exposing getPotentials (algorithm-specific result) and getMatchingWeight (exact BigDecimal instead of the double returned by the Matching interface), but isn't getMatchingEdges redundant with getMatching().getEdges()?